mongo/integration/unified/client_entity.go

```diff
@@ -133,21 +176,171 @@ func (c *clientEntity) failedEvents() []*event.CommandFailedEvent {
 	return events
 }
 
+func getCurrentTime() int64 {
+	return time.Now().Unix()
+}
+
 func (c *clientEntity) processStartedEvent(_ context.Context, evt *event.CommandStartedEvent) {
 	if c.getRecordEvents() {
```

Comment:
nit: Because there's a lot more logic in these functions than there previously was, we should refactor them to avoid additional indentation:

```
if !c.getRecordEvents() {
    return
}

// logic
```